apps/editing-toolkit/editing-toolkit-plugin/error-reporting/index.php

```diff
@@ -40,45 +48,95 @@ function add_crossorigin_to_script_els( $tag ) {
+ * activating it for WoA, too.
+ *
+ * @param int $user_id the user id.
+ * @return bool
+ */
+function user_in_sentry_test_segment( $user_id ) {
+	$current_segment = 1; // segment of existing users that will get this feature in %.
+	$user_segment    = $user_id % 100;
+
+	// We get the last two digits of the user id and that will be used to decide in what
+	// segment the user is. i.e if current_segment is 10, then only ids that end in < 10
+	// will be considered part of the segment.
+	return $user_segment < $current_segment;
+}
+
+/**
+ * Returns whether or not the site loading ETK is in the WoA env.
+ *
+ * @return bool
+ */
+function is_atomic() {
+	return defined( 'IS_ATOMIC' ) && IS_ATOMIC;
+}
+
+/**
+ * Return whether Sentry should be activated for a given user.
+ *
+ * In this phase, a12s have the possibility of configuring what error reporter to choose
+ * through the sticker. a12s shouldn not be covered by the segment logic.
+ *
+ * Regular users have the error reporter chosen based on the segmentation logic, only.
+ *
+ * @param int $user_id The user id. Used to check if the user is A8C or in
+ * the Sentry test segment.
+ * @param int $blog_id The blog ID. Usually the value of `get_current_blog_id`.
+ * Used to check if the sticker is applied if user is A8C.
+ */
+function should_activate_sentry( $user_id, $blog_id ) {
+	return ( is_automattician( $user_id ) && has_blog_sticker( 'error-reporting-use-sentry', $blog_id ) )
+		|| ( ! is_automattician( $user_id ) && user_in_sentry_test_segment( $user_id ) );
+}
+
 /**
  * Enqueue assets
  */
 function enqueue_script() {
 	$asset_file          = include plugin_dir_path( __FILE__ ) . 'dist/error-reporting.asset.php';
 	$script_dependencies = isset( $asset_file['dependencies'] ) ? $asset_file['dependencies'] : array();
 	$script_version      = isset( $asset_file['version'] ) ? $asset_file['version'] : filemtime( plugin_dir_path( __FILE__ ) . 'dist/error-reporting.js' );
+	$script_id           = 'a8c-fse-error-reporting-script';
 
 	wp_enqueue_script(
-		'a8c-fse-error-reporting-script',
+		$script_id,
 		plugins_url( 'dist/error-reporting.js', __FILE__ ),
 		$script_dependencies,
 		$script_version,
 		true
 	);
+
+	wp_localize_script(
+		$script_id,
+		'A8C_ETK_ErrorReporting_Config',
+		array(
+			'shouldActivateSentry' => should_activate_sentry( get_current_user_id(), get_current_blog_id() ) ? 'true' : 'false',
+		)
+	);
 }
 
 /**
- * Effectivelly activates the error reporting module by setting the necessary hooks.
+ * Setup the error reporting module by setting the necessary hooks. Whether or not we get the
+ * homebrew version or Sentry is decided inside the `enqueue_script` function, stored in a bool
+ * var, and passed over to the clientside script (index.js) which will then use this value to
+ * decide what tool to activate.
  */
-function activate_error_reporting() {
+function setup_error_reporting() {
 	add_action( 'admin_print_scripts', __NAMESPACE__ . '\head_error_handler' );
```

Comment:
Out of curiosity, would we want to set this to priority such that it is the first script executed by the browser? Otherwise, is there a chance that some scripts start getting executed (and throwing errors) before this is even available?

Reply:
That's a good point! Thanks for bringing that up. I guess a scenario where another header script is printed/enqueued before this one is possible 🤔. I'm not super familiar with how the priority algorithm works, but the [documentation](https://developer.wordpress.org/reference/functions/add_action/) for `add_action`'s `$priority` parameter says:

```
 (int) (Optional) Used to specify the order in which the functions associated with a particular action are executed. Lower numbers correspond with earlier execution, and functions with the same priority are executed in the order in which they were added to the action. 
```

So I'd say setting it to zero would do it?

Reply:
I'll set it to `0` and do some smoke testing.

Reply:
I saw no difference in the order the script element is added to the page when setting the priority to `0`, but I guess it's a good idea to leave it at it, given the default is `10`.

Reply:
Nice, that's good to know!